src/transformers/generation/utils.py

```diff
@@ -829,6 +829,46 @@ def _get_logits_warper(
             warpers.append(LogitNormalization())
         return warpers
 
+    def _get_generation_mode(
```

Comment:
It would be nice to have the different generation modes as an explicit enum, rather than strings. It'd make it clearer which modes are possible and easier to check `my_enum.VALUE` is less error prone that `"value"`

Reply:
Good suggestion 👍